### PR TITLE
Implement try catch react-services

### LINKED ISSUE #3435: Refactor try catch in React services
| Parent Issue |
|:---:|
|#3265| 

**Description**
In order to improve better error handling in the front-end, we should implement a better try-catch strategy in the `react-services` section, services, controller.

### PR DESCRIPTION
**Pull request by pablomarga**
Hi team,
This PR implement our new error handling strategy on the react services section

Close #3435

### CODE REVIEW

File: public/react-services/action-agents.js [hunk 1]
```diff
@@ -10,176 +10,219 @@
...
-        error !== 'Wazuh API error: 3021 - Timeout executing API request'
-          ? this.showToast('danger', 'Error upgrading agent', error, 5000)
-          : false;
+  static async upgradeAgent(agentId) {
+    try {
+      await WzRequest.apiReq('PUT', `/agents/${agentId}/upgrade`, {
+        force: 1,
       });
-    this.showToast('success', 'Upgrading agent...', '', 5000);
+      console.log('Upgrading');
```
Comment (Desvelao) [inline public/react-services/action-agents.js:37]: **nitpick**: Remove the `console.log`

File: public/react-services/action-agents.js [hunk 2]
```diff
@@ -10,176 +10,219 @@
...
+  static async upgradeAgents(selectedItems) {
     for (let item of selectedItems.filter(
-      item => item.outdated && item.status !== 'Disconnected'
+      (item) => item.outdated && item.status !== 'Disconnected'
     )) {
-      WzRequest.apiReq('PUT', `/agents/${item.id}/upgrade`, '1')
-        .then(() => {})
-        .catch(error => {});
+      try {
+        WzRequest.apiReq('PUT', `/agents/${item.id}/upgrade`, '1');
```
Comment (Desvelao) [inline public/react-services/action-agents.js:64]: **nitpick**: The `await` keyword is missing in the asynchronous method to wait for the response.

I am not sure if we need the `1` as the `body` parameter.

File: public/react-services/saved-objects.js [hunk 3]
```diff
@@ -216,7 +209,7 @@ export class SavedObject {
 
       return;
     } catch (error) {
-      console.log(error)
+      console.log(error);
```
Comment (Desvelao) [inline public/react-services/saved-objects.js:212]: **nitpick**: Remove the `console.error`.

File: public/react-services/vis-factory-handler.js [hunk 4]
```diff
@@ -79,7 +82,18 @@ export class VisFactoryHandler {
...
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.CRITICAL,
+          store: true,
+          error: {
+            error: error,
+            message: error.message || error,
+            title: error.name || error,
+          },
+        };
+        getErrorOrchestrator().handleError(options);
```
Comment (Desvelao) [inline public/react-services/vis-factory-handler.js:96]: **question**: The error was propagated before the change. Should it throw the error with the changes?

File: public/react-services/vis-factory-handler.js [hunk 5]
```diff
@@ -114,7 +128,18 @@ export class VisFactoryHandler {
...
```
Comment (Desvelao) [inline public/react-services/vis-factory-handler.js:142]: question: The error was propagated before the change. Should it throw the error with the changes?

File: public/react-services/check-daemons-status.js [hunk 6]
```diff
@@ -40,6 +40,7 @@ export class CheckDaemonsStatus {
       }
     } catch (error) {
       store.dispatch(updateWazuhNotReadyYet('Wazuh could not be recovered.'));
```
Comment (frankeros) [inline public/react-services/check-daemons-status.js:42]: it's taking a action by itself, is this correct?

File: public/react-services/group-handler.js [hunk 7]
```diff
@@ -10,70 +10,91 @@
...
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        store: true,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: `Error removing selected group`,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
```
Comment (frankeros) [inline public/react-services/group-handler.js:34]: why is the service calling to ErrorHandling?  It should be throw to the upper layers (?)

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark:

**Review by gabiwassan — APPROVED** [review]
LGTM

**Pull request merged**